modules/combined/test/tests/material_inversion/test_param_set/model_test.i

```diff
@@ -0,0 +1,169 @@
+[GlobalParams]
+  displacements = 'ux uy'
+[]
+
+[Mesh]
+  [gmg]
+    type = GeneratedMeshGenerator
+    dim = 2
+    nx = 11
+    ny = 11
+    xmin = -4
+    xmax = 4
+    ymin = -4
+    ymax = 4
+  []
+[]
+
+[AuxVariables]
+  [dummy]
+  []
+  [T]
+  []
+[]
+
+[Modules]
+  [TensorMechanics]
+    [Master]
+      [all]
+        strain = SMALL
+        new_system = true
+        add_variables = true
+        formulation = TOTAL
+        incremental = true
+        volumetric_locking_correction = false
+        generate_output = 'cauchy_stress_xx'
+      []
+    []
+  []
+[]
+
+[BCs]
+  [bottom_x]
+    type = DirichletBC
+    variable = ux
+    boundary = bottom
+    value = 0.0
+  []
+  [bottom_y]
+    type = DirichletBC
+    variable = uy
+    boundary = bottom
+    value = 0.0
+  []
+  [top_x]
+    type = NeumannBC
+    variable = ux
+    boundary = top
+    value = 1.0
+  []
+  [top_y]
+    type = NeumannBC
+    variable = uy
+    boundary = top
+    value = 1.0
+  []
+[]
+
+[NEML2]
+  input = 'elasticity.i'
+  model = 'elasticity_model'
+  temperature = 'T'
+  verbose = true
+  mode = PARSE_ONLY
+  device = 'cpu'
+[]
+
+[Materials]
+  [stress]
+    type = CauchyStressFromNEML2Receiver
+    neml2_uo = neml2_stress_UO
+  []
+  [E_material]
+    type = GenericFunctionMaterial
+    prop_names = 'E_material'
+    prop_values = E
+  []
+[]
+
+[Functions]
+  [E]
+    type = NearestReporterCoordinatesFunction
+    x_coord_name = parametrization/coordx
+    y_coord_name = parametrization/coordy
+    value_name = parametrization/youngs_modulus
+  []
+[]
+
+[Reporters]
+  [measure_data]
+    type = OptimizationData
+    variable = ux
+  []
+  [parametrization]
+    type = ConstantReporter
+    real_vector_names = 'coordx coordy youngs_modulus'
+    real_vector_values = '0 1 2; 0 1 2; 5 4 3'
+  []
+[]
```

Comment:
@hugary1995 here is where I want to change the `youngs_modulus`. The idea is a bit complicated: 
 - set/transfer the `youngs_modulus` values at three physical locations in this reporter
 - extrapolate it in the function using `NearestReporterCoordinatesFunction`
 - transform the function to a material property `E_material` 
 - transform the material property as a batch material UO `E_batch_material`
 - then the UO is consumed in the `CauchyStressFromNEML2UO` for updating the neml2 material model parameter
 
So by changing the values here in the reporter, ideally we should see different displacement values. But I do not see the values changing.. I have checked that the values in `E_batch_material` changes correspondingly as I change the value in this reporter, which indicates an issue in setting the parameters in the `CauchyStressFromNEML2UO`